src/Chapter2.hs

```diff
@@ -683,139 +551,110 @@ partially, we will get a new function that takes only one number and
 returns the result of dividing 10 by that number. You can check the
 difference by inspecting the types of corresponding expressions in
 GHCi:
-
 >>> :t +d div
 div :: Integer -> Integer -> Integer
 >>> :t +d div 10
 div 10 :: Integer -> Integer
-
-
 This fact can be used to pass partial applications of some functions
 directly to other functions.
-
 >>> map (div 10) [1 .. 10]
 [10,5,3,2,2,1,1,1,1,1]
-
 You can apply operators partially too!
-
 >>> filter (< 3) [2, 1, 3, 4, 0, 5]
 [2,1,0]
 >>> map (* 2) [1..5]
 [2,4,6,8,10]
-
 The implementation of the "map" function is pretty straightforward if
 you are already familiar with function application, recursion and
 pattern matching.
-
 @
 map :: (a -> b) -> [a] -> [b]
 map _ [] = []
 map f (x:xs) = f x : map f xs
 @
-
 Now you can see that there is nothing magic in HOFs in the end!
 -}
 
 {- |
 =⚔️= Task 8
-
 Implement a function that repeats each element so many times as the
 value of the element itself
-
 >>> smartReplicate [3, 1, 2]
 [3,3,3,1,2,2]
-
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate l = concat(map(\n -> replicate n n) l)
```

Comment:
Great! 👍🏼 
If you want to write this function even more elegantly, you can use the standard function `concatMap` which is a combination of `concat` and `map` as the name suggests 🙂
After that, you could notice, that you would be able to eta-reduce on the last argument :)